kind-setup.sh

```diff
@@ -25,7 +25,7 @@
 function get_status() {
     OBJECT=$1
 
-    kubectl get $OBJECT 2> /tmp/kubetctl.err | awk '
+    sudo kubectl get $OBJECT 2> ${KUBECTL_LOG} | awk '
```

Comment:
Currently in my environment "./kind-setup dev" works pretty well for latest code. I have concern after this pr, then all of our kubectl cmd needs to be under sudo?

Reply:
./kind-setup dev does not currently work if you are logged in as 'ubuntu' user. Are you logged in as root?

Reply:
I'm using ubuntu. It's working fine.

Reply:
Yeah it works for me too.
I think we should leave the sudo as a preference for user rather than put it in the script.
Over on Zeta we actually disallow all our scripts from running as sudo altogether.

Reply:
Thanks for the reviews. I have revised the change to remove sudo from all kubectl commands - it was not required. I've left it for docker and kind commands. The reason is that non-root user may not be auto-added to the docker group, and  will get permission denied errors unless you add the user to docker group and re-login.

Reply:
I'm not sure whether kind permission is related with docker. I have left another comments on the kind.

Reply:
So all of you are able to follow the instructions https://github.com/CentaurusInfra/mizar/blob/dev-next/docs/user/getting_started.md and it works for you out-of-box? 

I just tried a fresh 20.04 ubuntu VM in AWS, did bootstrap, and did ./kind-setup dev and hit various issues. Is it only me?

Reply:
This is what I get after following our getting_started guide:

ubuntu@ip-192-168-1-184:~/mizar$ ./kind-setup.sh dev
./kind-setup.sh: line 66: kind: command not found
Got permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Post http://%2Fvar%2Frun%2Fdocker.sock/v1.24/networks/create: dial unix /var/run/docker.sock: connect: permission denied
Got permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Post http://%2Fvar%2Frun%2Fdocker.sock/v1.24/build?buildargs=%7B%7D&cachefrom=%5B%5D&cgroupparent=&cpuperiod=0&cpuquota=0&cpusetcpus=&cpusetmems=&cpushares=0&dockerfile=k8s%2Fkind%2FDockerfile&labels=%7B%7D&memory=0&memswap=0&networkmode=default&rm=1&shmsize=0&t=localhost%3A5000%2Fkindnode%3Alatest&target=&ulimits=null&version=1: dial unix /var/run/docker.sock: connect: permission denied
k8s/kind/create_cluster.sh: line 34: kind: command not found
docker: Got permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Post http://%2Fvar%2Frun%2Fdocker.sock/v1.24/containers/create?name=local-kind-registry: dial unix /var/run/docker.sock: connect: permission denied.
See 'docker run --help'.
...
...

Reply:
This is what I did to get it to work as a non-root user
https://docs.docker.com/engine/install/linux-postinstall/

Reply:
I know to avoid sudo for docker, there needs to be groupadd for docker. And it's fair we don't assume user will do that. I'm just not sure whether we really need sudo for kind.

Reply:
Ok, so bottomline is that our instructions does not work for non-root user out of box and results in poor experience. They need enough familiarity with linux to take these additional steps to get things to work. Is that a correct assessment?

Reply:
And I also see that our documentation at readthedocs suggests using sudo, and we don't do so in our github docs. So we are inconsistent. Please see: https://mizar.readthedocs.io/en/latest/user/getting_started/

We should pick one route and be consistent about it. Let's discuss tomorrow.

Reply:
I'm fixed up the bootstrap script which we ask folks to run to handle these hidden steps. Please review.